pkg/chartverifier/checks/helm.go

```diff
@@ -200,27 +201,39 @@ func getImageReferences(chartUri string, vals map[string]interface{}) ([]string,
 	mem.SetNamespace("TestNamespace")
 	actionConfig.Releases = storage.Init(mem)
 
-	imagesMap := make(map[string]bool)
-
 	txt, err := actions.RenderManifests("testRelease", chartUri, vals, actionConfig)
+	if err != nil {
+		return nil, err
+	}
+
+	return getImagesFromContent(txt)
+
+}
+
+func getImagesFromContent(content string) ([]string, error) {
+
+	imagesMap := make(map[string]bool)
 
 	type ImageRef struct {
 		Ref string `yaml:"image"`
 	}
```

Comment:
Any particular reason this type is defined in the context of the enclosing function? I assume its content is serialized somewhere, as the `yaml` tag is being explicitly used.

Unless major changes are required, I'd also move this type outside the enclosing function.

Reply:
Not part is his fix so will leave as-is